### ISSUE TITLE
got exception when re-labeling entity while using interactive learning

### ISSUE THREAD

**Issue by carlhung** [RasaHQ/rasa#7185]
**Rasa version**: 2.0.0

**Rasa SDK version** (if used & relevant): 2.0.0

**Rasa X version** (if used & relevant): None

**Python version**: 3.8.6 (default, Oct  8 2020, 14:06:32) 

**Operating system** (windows, osx, ...): macOS-10.15.7-x86_64-i386-64bit

**Issue**:

when using interactive learning by command `rasa interactive`, rasa recognised a set of entities. i tried to relabel the entity. 

**Error (including full traceback)**:
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/rasa/core/training/interactive.py", line 1488, in record_messages
    await _validate_nlu(intents, endpoint, conversation_id)
  File "/usr/local/lib/python3.8/site-packages/rasa/core/training/interactive.py", line 1263, in _validate_nlu
    entities = await _correct_entities(latest_message, endpoint, conversation_id)
  File "/usr/local/lib/python3.8/site-packages/rasa/core/training/interactive.py", line 1288, in _correct_entities
    parse_annotated = entities_parser.parse_training_example(annotation)
  File "/usr/local/lib/python3.8/site-packages/rasa/shared/nlu/training_data/entities_parser.py", line 174, in parse_training_example
    entities = find_entities_in_training_example(example)
  File "/usr/local/lib/python3.8/site-packages/rasa/shared/nlu/training_data/entities_parser.py", line 52, in find_entities_in_training_example
    entity_attributes = extract_entity_attributes(match)
  File "/usr/local/lib/python3.8/site-packages/rasa/shared/nlu/training_data/entities_parser.py", line 84, in extract_entity_attributes
    return extract_entity_attributes_from_dict(entity_text, match)
  File "/usr/local/lib/python3.8/site-packages/rasa/shared/nlu/training_data/entities_parser.py", line 109, in extract_entity_attributes_from_dict
    entity_dict = get_validated_dict(entity_dict_str)
  File "/usr/local/lib/python3.8/site-packages/rasa/shared/nlu/training_data/entities_parser.py", line 151, in get_validated_dict
    validation_utils.validate_training_data(data, schema.entity_dict_schema())
  File "/usr/local/lib/python3.8/site-packages/rasa/shared/utils/validation.py", line 192, in validate_training_data
    raise e
  File "/usr/local/lib/python3.8/site-packages/rasa/shared/utils/validation.py", line 185, in validate_training_data
    validate(json_data, schema)
  File "/usr/local/lib/python3.8/site-packages/jsonschema/validators.py", line 934, in validate
    raise error
jsonschema.exceptions.ValidationError: 2 is not of type 'string'. Failed to validate data, make sure your data is valid. For more information about the format visit https://rasa.com/docs/rasa/training-data-format.

Failed validating 'type' in schema['properties']['value']:
    {'type': 'string'}

On instance['value']:
    2
2020-11-05 12:24:00 ERROR    asyncio  - Task exception was never retrieved
future: <Task finished name='Task-27' coro=<_serve_application.<locals>.run_interactive_io() done, defined at /usr/local/lib/python3.8/site-packages/rasa/core/training/interactive.py:1580> exception=<ValidationError: "2 is not of type 'string'. Failed to validate data, make sure your data is valid. For more information about the format visit https://rasa.com/docs/rasa/training-data-format.">>
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/rasa/core/training/interactive.py", line 1583, in run_interactive_io
    await record_messages(
  File "/usr/local/lib/python3.8/site-packages/rasa/core/training/interactive.py", line 1488, in record_messages
    await _validate_nlu(intents, endpoint, conversation_id)
  File "/usr/local/lib/python3.8/site-packages/rasa/core/training/interactive.py", line 1263, in _validate_nlu
    entities = await _correct_entities(latest_message, endpoint, conversation_id)
  File "/usr/local/lib/python3.8/site-packages/rasa/core/training/interactive.py", line 1288, in _correct_entities
    parse_annotated = entities_parser.parse_training_example(annotation)
  File "/usr/local/lib/python3.8/site-packages/rasa/shared/nlu/training_data/entities_parser.py", line 174, in parse_training_example
    entities = find_entities_in_training_example(example)
  File "/usr/local/lib/python3.8/site-packages/rasa/shared/nlu/training_data/entities_parser.py", line 52, in find_entities_in_training_example
    entity_attributes = extract_entity_attributes(match)
  File "/usr/local/lib/python3.8/site-packages/rasa/shared/nlu/training_data/entities_parser.py", line 84, in extract_entity_attributes
    return extract_entity_attributes_from_dict(entity_text, match)
  File "/usr/local/lib/python3.8/site-packages/rasa/shared/nlu/training_data/entities_parser.py", line 109, in extract_entity_attributes_from_dict
    entity_dict = get_validated_dict(entity_dict_str)
  File "/usr/local/lib/python3.8/site-packages/rasa/shared/nlu/training_data/entities_parser.py", line 151, in get_validated_dict
    validation_utils.validate_training_data(data, schema.entity_dict_schema())
  File "/usr/local/lib/python3.8/site-packages/rasa/shared/utils/validation.py", line 192, in validate_training_data
    raise e
  File "/usr/local/lib/python3.8/site-packages/rasa/shared/utils/validation.py", line 185, in validate_training_data
    validate(json_data, schema)
  File "/usr/local/lib/python3.8/site-packages/jsonschema/validators.py", line 934, in validate
    raise error
jsonschema.exceptions.ValidationError: 2 is not of type 'string'. Failed to validate data, make sure your data is valid. For more information about the format visit https://rasa.com/docs/rasa/training-data-format.

Failed validating 'type' in schema['properties']['value']:
    {'type': 'string'}

On instance['value']:
    2

```

**Command or request that led to error**:
```
rasa interactive
```

**Content of configuration file (config.yml)** (if relevant):
```yml
language: en

pipeline:
  - name: WhitespaceTokenizer
  - name: RegexFeaturizer
  - name: LexicalSyntacticFeaturizer
  - name: CountVectorsFeaturizer
  - name: CountVectorsFeaturizer
    analyzer: "char_wb"
    min_ngram: 1
    max_ngram: 4
  - name: DIETClassifier
    epochs: 100
  - name: DucklingEntityExtractor
    url: http://localhost:8000
    dimensions:
    - amount-of-money
    - time
    - number
  - name: "SpacyNLP"
    model: "en_core_web_md"
  - name: "SpacyEntityExtractor"
    dimensions: ["PERSON"]
  - name: EntitySynonymMapper # always after entity extractors
  - name: FallbackClassifier
    threshold: 0.5
    ambiguity_threshold: 0.1

policies:
  - name: AugmentedMemoizationPolicy
  - name: TEDPolicy
    epochs: 20 
  - name: RulePolicy
    core_fallback_threshold: 0.5
```

**Content of domain file (domain.yml)** (if relevant):
```yml
version: "2.0"

intents:
  - greet
  - goodbye
  - affirm
  - deny
  - mood_great
  - mood_unhappy
  - bot_challenge
  - want_to_order
  - nlu_fallback

responses:
  utter_greet:
  - text: "Hey! How are you?"

  utter_cheer_up:
  - text: "Here is something to cheer you up:"
    image: "https://i.imgur.com/nGF1K8f.jpg"

  utter_did_that_help:
  - text: "Did that help you?"

  utter_happy:
  - text: "Great, carry on!"

  utter_goodbye:
  - text: "Bye"

  utter_iamabot:
  - text: "I am a bot, powered by Rasa."

  utter_please_rephrase:
  - text: I'm sorry, I didn't quite understand that. Could you rephrase?

  utter_alright:
  - text: OK.

  utter_default:
  - text: "failed to understand. `action_default_fallback` was run."

  utter_ask_order_form_orderList:
  - text: What are the orders?
  - text: what would you like to order?

actions:
  - action_ask_order_form_numberList 
  - action_ask_order_form_unitList
  - action_check_order

forms:
  order_form:
    orderList:
    - type: from_entity
      entity: item
      intent: want_to_order
    unitList:
    - type: from_entity
      entity: unit
      intent: want_to_order
    numberList:
    - type: from_entity
      entity: number
      intent: want_to_order

entities:
  - item
  - amount-of-money
  - time
  - PERSON
  - number
  - unit

slots:
  orderList:
    type: list
  numberList:
    type: list
  unitList:
    type: list
  amount-of-money:
    type: text
  time:
    type: text
  PERSON:
    type: text

session_config:
  session_expiration_time: 60
  carry_over_slots_to_new_session: true
```

**Comment by sara-tagger** [comment]
Thanks for the issue, @samsucik will get back to you about it soon!

###### You may find help in the [docs](https://rasa.com/docs/) and the [forum,](https://forum.rasa.com/) too 🤗

**Comment by samsucik** [comment]
@carlhung thanks for reporting this, looks related to #7031 where unquoted numbers in JSON entity dictionaries (e.g. `[2]{"entity": "number", "value": 2}` were resulting in errors.

@degiz @kalkbrennerei could you have a look as it's likely the same or a very similar underlying issue?

**Comment by carlhung** [comment]
yes, i just ran again and test. if using duckling entity extractor, it extracts a value which is int.

**Comment by degiz** [comment]
Hey @carlhung, seems like https://github.com/RasaHQ/rasa/pull/7040 is fixing that issue.

**Comment by kalkbrennerei** [comment]
I'm closing this issue now since #7040 got merged
